js/src/elements/tab-container/element.ts

```diff
@@ -0,0 +1,471 @@
+		const afterSlot = document.createElement('slot');
+		afterSlot.setAttribute('part', 'after-panels');
+		afterSlot.setAttribute('name', 'after-panels');
+		shadowRoot.replaceChildren(tabListContainer, panelSlot, afterSlot);
+
+		if (this.#internals && 'role' in this.#internals) {
+			this.#internals.role = 'presentation';
+		} else {
+			this.setAttribute('role', 'presentation');
+		}
+
+		this.addEventListener('keydown', this);
+		this.addEventListener('click', this);
+
+		this.selectTab(-1);
+		this.#setupComplete = true;
+
+		this.initTabContainerChangeListener();
+		this.initTabContainerNavClickListener();
+	}
+
+	attributeChangedCallback(name: string) {
+		if (!this.isConnected || !this.shadowRoot) return;
+		if (name === 'vertical') {
+			this.vertical = this.hasAttribute('vertical');
+		}
+	}
+
+	handleEvent(event: Event) {
+		if (event.type === 'click') return this.#handleClick(event as MouseEvent);
+		if (event.type === 'keydown') return this.#handleKeydown(event as KeyboardEvent);
+	}
+
+	#handleKeydown(event: KeyboardEvent) {
+		const tab = (event.target as HTMLElement)?.closest?.('[role="tab"]');
+		if (!tab) return;
+		const tabs = this.#tabs;
+		if (!tabs.includes(tab as HTMLElement)) return;
+
+		const currentIndex = this.selectedTabIndex;
+		const vertical =
+			tab.closest('[role="tablist"]')?.getAttribute('aria-orientation') === 'vertical';
+		const prevTab = event.code === 'ArrowLeft' || (vertical && event.code === 'ArrowUp');
+		const nextTab = event.code === 'ArrowRight' || (vertical && event.code === 'ArrowDown');
+
+		if (nextTab) {
+			let index = currentIndex + 1;
+			if (index >= tabs.length) index = 0;
+			this.selectTab(index);
+		} else if (prevTab) {
+			let index = currentIndex - 1;
+			if (index < 0) index = tabs.length - 1;
+			this.selectTab(index);
+		} else if (event.code === 'Home') {
+			this.selectTab(0);
+			event.preventDefault();
+		} else if (event.code === 'End') {
+			this.selectTab(tabs.length - 1);
+			event.preventDefault();
+		}
+	}
+
+	#handleClick(event: MouseEvent) {
+		const tab = (event.target as HTMLElement)?.closest?.('[role=tab]');
+		if (!tab) return;
+		const tabs = this.#tabs;
+		const index = tabs.indexOf(tab as HTMLElement);
+		if (index >= 0) this.selectTab(index);
+	}
+
+	#reflectAttributeToShadow(name: string, node: Element) {
+		if (this.hasAttribute(name)) {
+			node.setAttribute(name, this.getAttribute(name)!);
+			this.removeAttribute(name);
+		}
+	}
+
+	get selectedTabIndex(): number {
+		return this.#tabs.findIndex(el => el.matches('[aria-selected=true]'));
+	}
+
+	set selectedTabIndex(i: number) {
+		this.selectTab(i);
+	}
+
+	get defaultTabIndex(): number {
+		return Number(this.getAttribute('default-tab') || -1);
+	}
+
+	set defaultTabIndex(index: number) {
+		this.setAttribute('default-tab', String(index));
+	}
+
+	initTabContainerChangeListener() {
+		this.addEventListener('tab-container-change', this.handleTabContainerChange.bind(this));
```

Comment:
Trying to fit conventions here:

```ts
	#scrollToTab(event: TabContainerChangeEvent) {
		const { tab } = event;
		if (!tab) {
			return;
		}

		(tab as HTMLElement).scrollIntoView({
			behavior: 'auto',
			block: 'nearest',
			inline: 'center'
		});
	}
```
Then in the constructor, I believe we'll have 

```ts
this.addEventListener('tab-container-change', this.scrollToTab);
```

Reply:
✅